tests/installation/first_boot.pm

```diff
@@ -27,6 +27,10 @@ sub run {
     my $boot_timeout = (get_var('SES5_DEPLOY') || check_var('VIRSH_VMM_FAMILY', 'hyperv')) ? 450 : 200;
     # SLE >= 15 s390x does not offer auto-started VNC server in SUT, only login prompt as in textmode
     return if check_var('ARCH', 's390x') && sle_version_at_least('15');
+    # Setup DM_NEEDS_USERNAME for SLE15 KDE migration case
+    # In SLES15 KDE has been drop, after migration the default desktop is XDM
+    # KDE has moved to Package Hub, it will stay install with SLES15 if added PackageHub
+    set_var('DM_NEEDS_USERNAME', '1') if (get_var('DESKTOP', 'KDE') && is_sle('15+') && (get_var('ADDONURL','') !~ /phub/));
```

Comment:
I doubt "first_boot" is a good location for this SLE specific hack. Would you consider moving this code to its own test module which you could call for example before "installation/reboot_after_installation"?

Even more so because "first_boot" is misleading as in the upgrade case it is actually not the "first boot".

Anyone has better ideas?

Reply:
How about move it to `version_switch_upgrade_target.pm` ?

Reply:
yes, looks like a very good place, great!